builtin/logical/database/secret_creds.go

```diff
@@ -91,24 +92,39 @@ func (b *databaseBackend) secretCredsRevoke() framework.OperationFunc {
 			return nil, fmt.Errorf("no role name was provided")
 		}
 
+		var dbName string
+		var statements dbplugin.Statements
+
 		role, err := b.Role(ctx, req.Storage, roleNameRaw.(string))
 		if err != nil {
 			return nil, err
 		}
-		if role == nil {
-			return nil, fmt.Errorf("error during revoke: could not find role with name %q", req.Secret.InternalData["role"])
+		if role != nil {
+			dbName = role.DBName
+			statements = role.Statements
+		} else {
+			if dbNameRaw, ok := req.Secret.InternalData["db_name"]; !ok {
```

Comment:
I wonder if we should just keep the current behavior of returning "role not found" if internal data does not have the fields.

Reply:
You mean, instead of the more specific error messages? Having those makes debugging easier because we know exactly where things are failing.